static/js/ui_init.js

```diff
@@ -106,7 +106,7 @@ function message_hover(message_row) {
 
     // Locally echoed messages have !is_topic_editable and thus go
     // through this code path.
-    if (!message_edit.is_topic_editable(message)) {
+    if (!message_edit.is_topic_editable(message) && message.type === "stream") {
```

Comment:
The main issue am worried about is this. Do you know something better for this check!

Reply:
How does this check if a message is locally echoed. We can just create a function for this.